src/realm/spec.cpp

```diff
@@ -59,34 +56,22 @@ void Spec::init(MemRef mem) noexcept
         m_enumkeys.detach();
     }
 
-    if (m_top.get_as_ref(s_col_keys_ndx) == 0) {
-        // This is an upgrade - create column key array
-        MemRef mem_ref = Array::create_empty_array(Array::type_Normal, false, m_top.get_alloc()); // Throws
-        m_keys.init_from_mem(mem_ref);
-        m_keys.update_parent();
-        size_t num_cols = m_types.size();
-        for (size_t i = 0; i < num_cols; i++) {
-            m_keys.add(i);
-        }
-    }
-    else {
-        m_keys.init_from_parent();
-    }
-
+    m_keys.init_from_parent();
```

Comment:
consider adding an assertion that m_top.get_as_ref(s_col_keys_ndx) is not zero

Reply:
Well, we don't check the other entries and I am sure we will have an assertion down the line if the ref is zero.